### ISSUE TITLE
Python: support batch dimension of length 1

### ISSUE THREAD

**Issue by WardBrian** [flatironinstitute/finufft#367]
Currently, I can call `finufft.nufft1d2(x,y,c)` where `c` has shape `(M,)` or `(ntrans, M)`, _except_ if `ntrans == 1`, in which case this raises a runtime error.

This makes extra work to plug into something like Pytorch, which expects to be able to batch with an arbitrary number of batches, including only 1 batch.

**Comment by janden** [comment]
Yes I think it makes sense to allow both `(1, M)` and `(M,)`, as long as the output has a compatible number of dimensions (i.e., in the first case the output has shape `(1, …)`). Question is if we want to differentiate between `ntrans = 1` and `ntrans = None`?

**Comment by ahbarnett** [comment]
Please go ahead with this python interface change, to allow `(1,M)` for input. I don't see a need to raise error if the output dims are then `(N,)`.  Re Joakim's question `ntrans=None` doesn't have a meaning for me (or for the C++ library)... the user wanting a single transform should set `ntrans=1`.  Libin, would you be able to implement this? Thanks, Alex

**Comment by lu1and10** [comment]
Yes, I will go ahead and deal this issue.

**Comment by janden** [comment]
Sorry I should have been more specific. By `ntrans = None` I mean that we would expect an input of size `(M,)` while `ntrans = 1` would mean we expect an input of size `(1, M)`. Or we can be more permissive and let `ntrans = 1` cover both cases (as long as we make sure the output has the correct number of dimensions).

**Comment by lu1and10** [comment]
@WardBrian We did some changes in PR#376 trying to support batch size 1 for CPU py interface. Could you check if it works on your side? Thanks!

**Comment by lu1and10** [comment]
> Seems to be working for us!

Thanks! I will close the issue for now. Let us know if you find any problem later.
